pages/_app.js

```diff
@@ -41,24 +45,39 @@ export default function App({ Component, pageProps }) {
     setPlantsInfo(plantsInfo.filter((plant) => plant.id !== id));
   }
 
+  function handleAddReminder(id, wateringDate) {
+    setPlantsInfo(
+      plantsInfo.map((plant) =>
+        plant.id === id ? { ...plant, wateringDate } : plant
+      )
+    );
+  }
+
   const isUserPlantFunction = (id) =>
     plantsInfo.find((info) => info.id === id)?.isUserPlant;
 
   const isMyPlantFunction = (id) =>
     plantsInfo.find((info) => info.id === id)?.isMyPlant;
 
+  const isMyPlantWithReminder = (id) =>
+    plantsInfo.find((info) => info.id === id)?.isMyPlant &&
+    !!plantsInfo.find((info) => info.id === id)?.wateringDate;
+
```

Comment:
this looks very similar to the isMyPlantFunction and isUserPlantFunction. You could simplify all three functions by creating a function called getPlantInfoById which returns the info Object for the respective id. Then you can use the object directly to determine if isUserPlant, isMyPlant or wateringDate exist on it. Then you need to pass only one function down to your Components.